lib/surface/compiler/parse_tree_translator.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Surface.Compiler.ParseTreeTranslator do
+  @behaviour Surface.Compiler.NodeTranslator
+
+  alias Surface.IOHelper
+
+  def handle_init(state), do: state
+
+  def handle_expression(state, expression, meta) do
+    {state, {:expr, expression, to_meta(meta)}}
+  end
+
+  def handle_comment(state, comment, meta) do
+    {state, {:comment, comment, meta}}
+  end
+
+  def handle_node(state, context, "template", attributes, body, meta) do
```

Comment:
I wonder if it would be better to have the node info at the beginning and move `state` and `context` to be the last arguments.

Reply:
I also flipped `{state, node}` to `{node, state}`; lmk what you think